app/lib/features/news/providers/news_post_editor_providers.dart

```diff
@@ -132,7 +134,18 @@ class NewsStateNotifier extends StateNotifier<NewsPostState> {
         state = state.copyWith(currentNewsSlide: selectedNewsSlide);
       },
     );
+  }
 
+  Future<void> selectInvitationCodeToShare(BuildContext context) async {
+    final superInviteCode = await selectSuperInviteListDrawer(context: context);
+    RefDetails? refDetails;
+    if (superInviteCode != null) {
+      final selectedInviteCode = await ref.watch(superInviteTokenProvider(superInviteCode.token()).future);
+      refDetails = selectedInviteCode.refDetails();
+    }
```

Comment:
`final selectedInviteCode = await ref.watch(superInviteTokenProvider(superInviteCode.token()).future);`

What is the need of this? 
As you are getting `SuperInviteToken` object directly from `selectSuperInviteListDrawer`

```suggestion
    final superInviteToken = await selectSuperInviteListDrawer(context: context);
    RefDetails? refDetails;
    if (superInviteToken != null) {
      refDetails = superInviteToken.refDetails();
    }
```